### PR TITLE
feat: delete keycloak user when deleting user entity

### PR DESCRIPTION
**Pull request by tomwwinter**
closes: https://github.com/Aam-Digital/ndb-core/issues/1874

### Visible/Frontend Changes

- [ ] show warning popup, when keycloak user could not removed after user deletion 

### Architectural/Backend Changes

- [ ] see https://github.com/Aam-Digital/account-backend/pull/38

### CODE REVIEW

File: src/app/core/entity-list/duplicate-records/duplicate-records.service.spec.ts [hunk 1]
```diff
@@ -41,6 +44,7 @@ describe("DuplicateRecordsService", () => {
         { provide: MatDialog, useValue: {} },
         { provide: MatSnackBar, useValue: {} },
         { provide: FileService, useValue: {} },
+        { provide: KeycloakAuthService, useValue: mockAuthService },
```
Comment (sleidig) [inline src/app/core/entity-list/duplicate-records/duplicate-records.service.spec.ts:47]: does this really need the service?

File: src/app/core/entity/entity-actions/entity-actions.service.ts [hunk 2]
```diff
@@ -102,9 +104,22 @@ export class EntityActionsService {
...
+          next: () => {},
+          error: async () =>
+            await this.confirmationDialog.getConfirmation(
+              $localize`:delete account in keycloak related error title:Keycloak User could not be deleted`,
+              $localize`:delete account in keycloak related error dialog:User Account could not be deleted in Keycloak. Please delete user manually in Keycloak.`,
+              OkButton,
+            ),
+        });
+      }
       result.mergeResults(await this.entityDelete.deleteEntity(entity));
```
Comment (sleidig) [inline src/app/core/entity/entity-actions/entity-actions.service.ts:120]: wouldn't this code logically rather belong in the EntityDeleteService, one layer down?

File: src/app/core/entity/entity-actions/entity-actions.service.ts [hunk 3]
```diff
@@ -102,9 +104,22 @@ export class EntityActionsService {
...
     let result = new CascadingActionResult();
+
     for (let entity of entities) {
+      if ("User" === entity.getType()) {
+        this.keycloakAuthService.deleteUser(entity.getId()).subscribe({
+          next: () => {},
+          error: async () =>
+            await this.confirmationDialog.getConfirmation(
+              $localize`:delete account in keycloak related error title:Keycloak User could not be deleted`,
+              $localize`:delete account in keycloak related error dialog:User Account could not be deleted in Keycloak. Please delete user manually in Keycloak.`,
```
Comment (sleidig) [inline src/app/core/entity/entity-actions/entity-actions.service.ts:115]: What happens if there actually is no user to be deleted? Do we get feedback from the API?

Comment (tomwwinter) [reply src/app/core/entity/entity-actions/entity-actions.service.ts:115]: Nothing, it's save to call this endpoint multiple times. If something there, we delete it, if not, nothing to delete - also fine.

File: src/app/core/entity-list/duplicate-records/duplicate-records.service.spec.ts (hunk 1 above)
Comment (tomwwinter) [reply src/app/core/entity-list/duplicate-records/duplicate-records.service.spec.ts:47]: Yes, some other dependency needs Keycloak.

**Review by sleidig — APPROVED** [review]

**Pull request merged**